### PR TITLE
MLDB-1861: added support for the "Expect: 100-continue" header

### PR DESCRIPTION
**Pull request by wsourdeau**

### CODE REVIEW

File: http/http_socket_handler.cc [hunk 1]
```diff
@@ -14,6 +15,14 @@
 using namespace std;
 using namespace Datacratic;
 
+namespace {
+
+const char header100Continue[] = "Expect: 100-continue\r\n";
+const int sizeof100Continue = sizeof(header100Continue) - 1;
```
Comment (guyd) [inline http/http_socket_handler.cc]: Can't you use constexpr here and on the above line?

File: http/http_socket_handler.cc [hunk 2]
```diff
@@ -219,6 +231,34 @@ onHeader(const char * data, size_t size)
...
+    }
+
+    return result;
+}
+
+bool
+HttpLegacySocketHandler::
+shouldReturn100Continue(const HttpHeader & header)
+{
+    return true;
```
Comment (guyd) [inline http/http_socket_handler.cc]: Are you missing some validation here?

File: http/http_socket_handler.h [hunk 3]
```diff
@@ -121,15 +125,24 @@ struct HttpLegacySocketHandler : public HttpSocketHandler {
               NextAction action = NEXT_CONTINUE,
               OnWriteFinished onWriteFinished = nullptr);
 
+protected:
+    /* Overridable method returning whether the given request should be
+       accepted for processing or not. The default implementation returns
+       "true". */
+    bool shouldReturn100Continue(const HttpHeader & header);
```
Comment (guyd) [inline http/http_socket_handler.h]: virtual I guess

Comment (wsourdeau) [inline http/http_socket_handler.h]: INdeed!

**Comment by guyd** [comment]
+1 with minor comments

**Pull request merged**